Components/ImportService.php

```diff
@@ -240,29 +256,7 @@ public function importRemoteCategory($localCategoryId, $remoteCategoryKey, $remo
         ];
 
         // create same category structure as Shopware Connect structure
-        $categories = $this->autoCategoryResolver->convertTreeToEntities($remoteCategoryNodes, $localCategory);
-
-        foreach ($categories as $category) {
-            $articleIds = $this->productToRemoteCategoryRepository->findArticleIdsByRemoteCategory($category['categoryKey']);
-
-            while ($currentIdBatch = array_splice($articleIds, 0, 10)) {
-                $articles = $this->articleRepository->findBy(['id' => $currentIdBatch]);
-                /** @var \Shopware\Models\Article\Article $article */
-                foreach ($articles as $article) {
-                    /** @var \Shopware\Models\Category\Category $categoryModel */
-                    $categoryModel = $category['model'];
-                    if ($article->getCategories()->contains($categoryModel->getParent())) {
```

Comment:
We have to keep this logic as well, because this is needed when some customers assign their products to categories which are not leaf. For example Mr.Strauss need this.

I'm not sure but now you assign all categories. That means if product comes with
following categories: 
How it looks like in product struct
```
$product->categories = [
'category1' => 'Category 1',
'category1/category2' => 'Category2',
'category1/category2/category3' => 'Category3'
]
```
How it looks like as a tree
Category 1 
----- Category 2 
----- ----- Category 3

After product import it will be assigned to Category1, Category2, Category3 and these assignments will be visible in shopware backend. Have you tested this?